uefi/src/proto/media/load_file.rs

```diff
@@ -0,0 +1,86 @@
+//! Load file support protocols.
+
+#[cfg(feature = "alloc")]
+use alloc::vec::Vec;
+
+use crate::proto::device_path::{FfiDevicePath, DevicePath};
+use crate::proto::unsafe_protocol;
+use crate::{Result, Status};
+use core::ffi::c_void;
+use core::ptr;
+
+/// The UEFI LoadFile2 protocol.
+///
+/// This protocol has a single method to load a file according to some
+/// device path.
+///
+/// This interface is implemented by many devices, e.g. network and filesystems.
+#[derive(Debug)]
+#[repr(C)]
+#[unsafe_protocol("4006c0c1-fcb3-403e-996d-4a6c8724e06d")]
+pub struct LoadFile2Protocol {
+    load_file: unsafe extern "efiapi" fn(
+        this: &mut LoadFile2Protocol,
+        file_path: *const FfiDevicePath,
+        boot_policy: bool,
+        buffer_size: *mut usize,
+        buffer: *mut c_void,
+    ) -> Status,
+}
+
+impl LoadFile2Protocol {
+    /// Load file addressed by provided device path
+    pub fn load_file(&mut self,
+        file_path: &DevicePath,
+        boot_policy: bool,
+        buffer_size: &mut usize,
+        buffer: *mut c_void
+    ) -> Status {
+        unsafe {
+            (self.load_file)(self,
+                file_path.as_ffi_ptr(),
+                boot_policy,
+                buffer_size as *mut usize,
+                buffer
+            )
+        }
+    }
+
+    #[cfg(feature = "alloc")]
+    /// Load file addressed by the provided device path.
+    pub fn load_file_in_heap(&mut self,
```

Comment:
For the name, we don't have any other functions with `heap` in the name (and the std lib also doesn't, except for `heapsort`). How about `load_file_to_vec`, mirroring `read_to_string`.

Reply:
Sounds good to me!

Reply:
Addressed.